BraveSharedTests/CustomHeaderDataTests.swift

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import XCTest
+@testable import BraveShared
+import SwiftyJSON
+
+class CustomHeaderDataTest: XCTestCase {
+    
+    //Valid Json from server.
+    var expectedJSON = "[{\"domains\":[\"coinbase.com\",\"api.coinbase.com\"],\"headers\":{\"X-Brave-Partner\":\"coinbase\"},\"cookieNames\":[],\"expiration\":31536000000},{\"domains\":[\"marketwatch.com\",\"barrons.com\"],\"headers\":{\"X-Brave-Partner\":\"dowjones\"},\"cookieNames\":[],\"expiration\":31536000000},{\"domains\":[\"townsquareblogs.com\",\"tasteofcountry.com\",\"ultimateclassicrock.com\",\"xxlmag.com\",\"popcrush.com\"],\"headers\":{\"X-Brave-Partner\":\"townsquare\"},\"cookieNames\":[],\"expiration\":31536000000},{\"domains\":[\"cheddar.com\"],\"headers\":{\"X-Brave-Partner\":\"cheddar\"},\"cookieNames\":[],\"expiration\":31536000000}]"
```

Comment:
Use `"""`, then you don't need to escape all of the `"`s

var expectedJSON = """[{"domains":["coinbase.com"]}..."""`

Reply:
Let's use completely fabricated data here, and not real websites/services, since this is all hardcoded (even though this is a test, still will stay in our code for people to see).

Reply:
Done